extensions/amp-ad/0.1/test/test-amp-ad.js

```diff
@@ -89,33 +90,62 @@ describe('A4A loader', () => {
         });
       });
 
-      it('adds network-specific child for registered, A4A type', () => {
+      it('upgrades to registered, A4A type network-specific element', () => {
         return createIframePromise().then(fixture => {
           const doc = fixture.doc;
+          const extensionsMock = sandbox.mock(extensionsFor(fixture.win));
           a4aRegistry['zort'] = function() {
             return true;
           };
           const element = doc.createElement(tag);
           element.setAttribute('type', 'zort');
-          element.setAttribute('width', '300');
-          element.setAttribute('height', '200');
           doc.body.appendChild(element);
+          const zortInstance = {};
+          const zortConstructor = function() { return zortInstance; };
+          extensionsMock.expects('loadElementClass')
+            .withExactArgs('amp-ad-network-zort-impl')
+            .returns(Promise.resolve(zortConstructor)).once();
           const handler = new AmpAd(element);
-          expect(handler.upgradeCallback()).to.be.null;
-          handler.buildCallback();
-          const expectedChild = childElement(element,
-              c => {
-                return c.tagName.indexOf('ZORT') >= 0;
-              });
-          expect(expectedChild).to.not.be.null;
-          expect(childElement(element,
-              c => {
-                return c.tagName === 'AMP-AD-3P-IMPL';
-              })).to.be.null;
-          expect(expectedChild).to.not.be.null;
-          expect(expectedChild.getAttribute('type')).to.equal('zort');
-          expect(expectedChild.getAttribute('width')).to.equal('300');
-          expect(expectedChild.getAttribute('height')).to.equal('200');
+          const upgradedElementPromise = handler.upgradeCallback();
+          extensionsMock.verify();
+          expect(upgradedElementPromise).not.to.be.null;
+          expect(element.getAttribute(
+            'data-a4a-upgrade-type', 'amp-ad-network-zort-impl'));
+          return upgradedElementPromise.then(baseElement => {
+            expect(
+              doc.head.querySelector('script[src*="amp-ad-network-zort-impl"]'))
+              .to.not.be.null;
+            expect(baseElement).to.equal(zortInstance);
+          });
+        });
+      });
+
+      it('falls back to 3p impl on upgrade with loadElementClass error', () => {
+        return createIframePromise().then(fixture => {
```

Comment:
can we try to extract element creation code to `beforeEach()` so we don't repeat them in each test case?